app/assets/javascripts/exchanges/carriers.js

```diff
@@ -68,3 +68,14 @@ $(document).on('turbolinks:load', function() {
         $('.plans-count').text(visiblePlansCount);
     }
 });
+
+$(document).on('turbolinks:request-start', function() {
+    $('#exchanges-loading-spinner').show();
+});
+
+$(document).on('turbolinks:load', function() {
+    $('#exchanges-loading-spinner').hide();
+});
+$(document).ready(function() {
```

Comment:
I could be misinterpreting, but I thought `turbolinks:load` handler was enough in recent Rails versions - https://github.com/turbolinks/turbolinks?tab=readme-ov-file#full-list-of-events:~:text=turbolinks%3Aload%20fires%20once%20after%20the%20initial%20page%20load%2C%20and%20again%20after%20every%20Turbolinks%20visit.%20Access%20visit%20timing%20metrics%20with%20the%20event.data.timing%20object.  So perhaps could drop `$(document).ready(...`?

Interested to learn if you've seen differently.  (`'ready turbolinks:load'` _might_ also be an option to reduce code duplication, if not)